tests/test-widgets/src/title.spec.ts

```diff
@@ -307,94 +307,44 @@ describe('@phosphor/widgets', () => {
-        title.changed.connect((sender, arg) => {
-          expect(sender).to.equal(title);
-          expect(arg).to.equal(undefined);
-          called = true;
-        });
-        title.setData('foo', '3');
-        expect(called).to.equal(true);
-      });
-
-      it('should not emit the changed signal when a value does not change', () => {
-        let called = false;
-        let title = new Title({ owner, dataset: { foo: '1', bar:  '2' }});
-        title.changed.connect((sender, arg) => {
-          expect(sender).to.equal(title);
-          expect(arg).to.equal(undefined);
-          called = true;
-        });
-        title.setData('foo', '1');
-        expect(called).to.equal(false);
+      it('should initialize from the options', () => {
+        let title = new Title({ owner, dataset: { foo: '12' } });
+        expect(title.dataset).to.deep.equal({ foo: '12' });
       });
 
-    });
-
-    describe('updateData()', () => {
-
-      it('should update several values in the dataset', () => {
-        let title = new Title({ owner, dataset: { foo: '1', bar:  '2' }});
-        title.updateData({ foo: '3', bar: '4', baz: '5' });
-        expect(title.getData('foo')).to.equal('3');
-        expect(title.getData('bar')).to.equal('4');
-        expect(title.getData('baz')).to.equal('5');
+      it('should be writable', () => {
+        let title = new Title({ owner, dataset: { foo: '12' } });
+        title.dataset = { bar: '42' };
+        expect(title.dataset).to.deep.equal({ bar: '42' });
       });
 
-      it('should emit the changed signal when any value changes', () => {
+      it('should emit the changed signal when the value changes', () => {
         let called = false;
-        let title = new Title({ owner, dataset: { foo: '1', bar:  '2' }});
+        let title = new Title({ owner, dataset: { foo: '12' } });
         title.changed.connect((sender, arg) => {
           expect(sender).to.equal(title);
           expect(arg).to.equal(undefined);
           called = true;
         });
-        title.updateData({ foo: '3', bar: '2' });
-        expect(called).to.equal(true);
-        called = false;
-        title.updateData({ foo: '3', bar: '2', baz: '5' });
+        title.dataset = { bar: '42' };
         expect(called).to.equal(true);
       });
 
-      it('should not emit the changed signal when no values change', () => {
+      it('should not emit the changed signal when the value does not change', () => {
         let called = false;
-        let title = new Title({ owner, dataset: { foo: '1', bar:  '2' }});
+        let dataset = { foo: '12' };
+        let title = new Title({ owner, dataset });
         title.changed.connect((sender, arg) => {
-          expect(sender).to.equal(title);
-          expect(arg).to.equal(undefined);
           called = true;
         });
-        title.updateData({ foo: '1', bar: '2' });
+        title.dataset = dataset;
```

Comment:
It will notify anyway. The consumer is typically rendering with vdom, so it won't really matter in the grand scheme of things.